vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/bdd/steps/ApplicationSteps.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.mobileapp.bdd.steps;
+
+import java.util.List;
+
+import org.jbehave.core.annotations.Given;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.mobileapp.bdd.model.DesiredCapability;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.selenium.SauceLabsCapabilityType;
+import org.vividus.selenium.manager.IWebDriverManagerContext;
+import org.vividus.selenium.manager.WebDriverManagerParameter;
+
+import io.appium.java_client.HasSessionDetails;
+
+public class ApplicationSteps
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationSteps.class);
+
+    private final IWebDriverProvider webDriverProvider;
+    private final IWebDriverManagerContext webDriverManagerContext;
+
+    public ApplicationSteps(IWebDriverProvider webDriverProvider, IWebDriverManagerContext webDriverManagerContext)
+    {
+        this.webDriverProvider = webDriverProvider;
+        this.webDriverManagerContext = webDriverManagerContext;
+    }
+
+    @Given("I run the application located at `$location` and capabilities:$capabilities")
```

Comment:
1) `Given("I start mobile application located at `$location`")`?
2) Why are capabilities needed?

Reply:
- Although we have an opportunity to set desired capabilities thorough meta at story level, by setting desired capabilities at step level we could use it in conjunction with lifecycle ET to run the same story several times on different devices
- Imho this is more clear way to setup properties than at meta level

Reply:
would not it better to name it
Given I start mobile application with capabilities: capabilities
?